torch_frame/nn/encoder/stype_encoder.py

```diff
@@ -819,7 +819,18 @@ def encode_forward(
                     for key in feat
                 })
             else:
-                x = self.model_dict[col_name](feat[:, i])
+                input_feat = feat[:, i]
+                if isinstance(input_feat, MultiEmbeddingTensor):
+                    input_feat = input_feat.values
+
+                # Numerical case:
+                if input_feat.ndim == 1:
+                    input_feat = input_feat.view(-1, 1, 1)
+                # MultiNestedEmbedding case:
+                elif input_feat.ndim == 2:
+                    input_feat = input_feat.unsqueeze(1)
+
+                x = self.model_dict[col_name](input_feat)
```

Comment:
Can you assert on the shape of `input_feat`?

Reply:
For categorical stype, asserting to ndim = 3 seems not reasonable.

Reply:
Why? I think it's fine

Reply:
The input_feat here for categorical stype is the index right? Reshaping it to a 3-dimensional tensor seems weird

Reply:
Yes, but the specification of model is to map `[batch_size, 1, in_channels]` to `[batch_size, 1, out_channels]`, so it sounds ok to me.

Reply:
where the middle "1" represents the single column.

Reply:
Ok